### PR TITLE
Metadata handling

### PR DESCRIPTION
**Pull request by matkrak**
Changes in both curve.py and profile.py that provide metadata handling

### CODE REVIEW

File: beprof/curve.py [hunk 1]
```diff
@@ -31,8 +31,12 @@ class Curve(np.ndarray):
     Has additional field - info.
     """
 
-    def __new__(cls, input_array):
+    def __new__(cls, input_array, **meta):
         obj = np.asarray(input_array).view(cls).copy()
+        if meta is None:
+            obj.metadata = {}
+        else:
+            obj.metadata = meta.copy()
```
Comment (grzanka) [inline beprof/curve.py]: I see you used shallow copy. Do you think it is better here than deep copy (https://docs.python.org/3.5/library/copy.html) ?

File: beprof/curve.py [hunk 2]
```diff
@@ -31,8 +31,12 @@ class Curve(np.ndarray):
     Has additional field - info.
     """
 
-    def __new__(cls, input_array):
+    def __new__(cls, input_array, **meta):
         obj = np.asarray(input_array).view(cls).copy()
```
Comment (grzanka) [inline beprof/curve.py]: Copying on the view here is also a subject of discussion. It doesn't necessarily need to stay here

File: beprof/curve.py (hunk 1 above)
Comment (matkrak) [inline beprof/curve.py]: Deep copy is what I wanted to do, my bad

File: beprof/curve.py (hunk 2 above)
Comment (matkrak) [inline beprof/curve.py]: Works without .copy() but that was something you wrote so I didn't dare to remove it!

**Comment by matkrak** [comment]
If you don't have any other comments, shall I rewrite tests as docstring and issue can be closed?

File: beprof/curve.py [hunk 3]
```diff
@@ -68,10 +73,44 @@ def y_at_x(self, x):
...
     def __str__(self):
         ret = "shape: {}".format(self.shape) + \
-              " X : [{:4.3f},{:4.3f}]".format(min(self.x), max(self.x)) + \
-              " Y : [{:4.6f},{:4.6f}]".format(min(self.y), max(self.y))
+              "\nX : [{:4.3f},{:4.3f}]".format(min(self.x), max(self.x)) + \
+              "\nY : [{:4.6f},{:4.6f}]".format(min(self.y), max(self.y)) + \
+              "\nMetadata : " + str(self.metadata)
         return ret
 
+    # functions not ready yet, added for testing metadata purpose
```
Comment (grzanka) [inline beprof/curve.py]: Remove it from this PR. It doesn't have anything in common with metadata handling.

**Comment by matkrak** [comment]
Should I add any docstring tests here? Didn't add any new method after all. I should add some text to docstring of Curve anyway - not a word about metadata for now.

**Comment by grzanka** [comment]
Add some docstring to Curve class describing how does it work.

**Pull request merged**